File: app/src/main/java/com/robivan/simplenote/NoteSource.java
package com.robivan.simplenote;

public interface NoteSource {
    NoteSource init(NoteSourceResponse noteSourceResponse);

    NoteEntity getNoteData(int position);

    int size();

    void deleteNoteData(int position); //TODO реализовать удаление

    void updateNoteData(NoteEntity noteData, int position);

    void addNoteData(NoteEntity noteData);

    void clearNoteData(); //TODO реализовать очистку
}

File: app/src/main/java/com/robivan/simplenote/NoteSource.kt
package com.robivan.simplenote

interface NoteSource {
    fun init(noteSourceResponse: NoteSourceResponse): NoteSource
    fun getNoteData(position: Int): NoteEntity?
    fun size(): Int
    fun deleteNoteData(position: Int)
    fun updateNoteData(noteData: NoteEntity?, position: Int)
    fun addNoteData(noteData: NoteEntity?)
    fun clearNoteData() //TODO реализовать очистку
}